tests/test_conf.py

```diff
@@ -0,0 +1,111 @@
+"""
+Tests the conf.py module.
+"""
+import os
+import sys
+import runpy
+from unittest import mock
+import pytest
+
+
+def test_conf_module_load():
+    from clustermatch import conf
+
+    assert conf is not None
+    assert conf.__file__ is not None
+
+
+def test_conf_entries():
+    from clustermatch import conf
+
+    assert conf.ROOT_DIR is not None
+    assert conf.ROOT_DIR != ""
+
+    assert conf.DATA_DIR is not None
+    assert conf.DATA_DIR != ""
+
+    assert conf.RESULTS_DIR is not None
+    assert conf.RESULTS_DIR != ""
+
+    assert conf.GENERAL is not None
+    assert len(conf.GENERAL) > 0
+    assert conf.GENERAL["N_JOBS"] is not None
+    assert conf.GENERAL["N_JOBS"] > 0
+    assert conf.GENERAL["N_JOBS_LOW"] is not None
+    assert conf.GENERAL["N_JOBS_LOW"] > 0
+
+    assert conf.RESULTS["BASE_DIR"] is not None
+    assert conf.RESULTS["BASE_DIR"] != ""
+
+    assert conf.MANUSCRIPT is not None
+
+
+def test_conf_main():
+    t = runpy.run_module("clustermatch.conf", run_name="__main__")
+    assert t is not None
+    assert "print_vars" in t
+    assert "CM_ROOT_DIR" in t["print_vars"]
+    assert "CM_RESULTS_DIR" in t["print_vars"]
+    assert "CM_GENERAL_N_JOBS" in t["print_vars"]
+
+
+@pytest.mark.skipif(
+    sys.platform.startswith("win"),
+    reason="exporting variables is only supported in non-Windows platforms",
+)
+def test_conf_export_variables():
+    from pathlib import Path
+    import subprocess
+    from clustermatch import conf
+
+    conf_filepath = Path(conf.__file__).resolve()
+    assert conf_filepath is not None
+    assert conf_filepath.exists()
+
+    # check output
+    r = subprocess.run(["python", conf_filepath], stdout=subprocess.PIPE)
+    assert r is not None
+    assert r.returncode == 0
+    r_output = r.stdout.decode("utf-8")
+    assert r_output is not None
+    assert len(r_output) > 10
+    assert r_output.count("export ") > 5
+
+    # check variable
+    r = subprocess.run(
+        f"eval `python {conf_filepath}` && echo $CM_ROOT_DIR",
+        shell=True,
+        stdout=subprocess.PIPE,
+    )
+    assert r is not None
+    assert r.returncode == 0
+    r_output = r.stdout.decode("utf-8").strip()
+    assert r_output is not None
+    assert len(r_output) > 10
+    assert r_output.startswith("/")
+
+    # check dict variable
+    r = subprocess.run(
+        f"eval `python {conf_filepath}` && echo $CM_GENERAL_N_JOBS",
+        shell=True,
+        stdout=subprocess.PIPE,
+    )
+    assert r is not None
+    assert r.returncode == 0
+    r_output = r.stdout.decode("utf-8").strip()
+    assert r_output is not None
+    assert r_output.isdigit()
+    assert int(r_output) > 0
+
+
+@mock.patch.dict(os.environ, {"CM_MANUSCRIPT_DIR": "/tmp/some/dir"})
+def test_conf_with_manuscript_dir():
```

Comment:
Do you want there to always be a manuscript dir? Would there ever be a case where you don't have one and you're okay with this test failing?

Or would the call to `mock.patch.dict` handle that? (I'm not entirely clear on what that decorator is doing)

Reply:
By default, `conf.py` does not have a manuscript folder value, so the rest of the tests handle the case you are describing (figures and tables just won't be generated). This particular case handles the situation where you do have a manuscript folder set (via environment variable, here injected with `mock`). So here I test if some variables are set (they should, since the manuscript folder was specified).

I've added the inverse check in the other tests to make sure those variables (`conf.MANUSCRIPT["CONTENT_DIR"]`) are not set when no manuscript dir was specified.